apm-agent-plugins/apm-logging-plugin/apm-jul-plugin/src/main/java/co/elastic/apm/agent/jul/reformatting/JulEcsReformattingHelper.java

```diff
@@ -86,7 +87,12 @@ protected String getAppenderName(StreamHandler handler) {
     protected Formatter createEcsFormatter(String eventDataset, @Nullable String serviceName, @Nullable String serviceVersion,
                                            @Nullable String serviceNodeName, @Nullable Map<String, String> additionalFields,
                                            Formatter originalFormatter) {
-        EcsFormatter ecsFormatter = new EcsFormatter();
+        EcsFormatter ecsFormatter = new EcsFormatter(){
+            @Override
+            protected Map<String, String> getMdcEntries() {
+                return AgentMDC.getEntries();
+            }
+        };
```

Comment:
[for reviewer] this is the reason we need to allow this through subclassing in ecs-logging-java: https://github.com/elastic/ecs-logging-java/pull/193